src/packet/sym_encrypted_integrity_protected_data.js

```diff
@@ -204,17 +212,22 @@ class SymEncryptedIntegrityProtectedDataPacket {
         if (!util.equalsUint8Array(hash, mdc)) {
           throw new Error('Modification detected.');
         }
+        // this last chunk signals the end of the stream to Packetlist.read's streamTransformPair .
```

Comment:
It's not this chunk that signals the end of the stream; the end of the stream is signaled explicitly. It's just that this chunk comes last in the stream due to the call to `streamConcat` below.
```suggestion
```